signer-service/src/api/services/stellar.service.js

```diff
@@ -151,4 +152,21 @@ async function sendStatusWithPk() {
   }
 }
 
-module.exports = { buildCreationStellarTx, buildPaymentAndMergeTx, sendStatusWithPk };
+async function signSep10Challenge(challengeXDR, outToken) {
+  const keypair = Keypair.fromSecret(CLIENT_SECRET);
+
+  const { signingKey } = await fetchTomlValues(TOKEN_CONFIG[outToken].tomlFileUrl);
+
+  const transactionSigned = new TransactionBuilder.fromXDR(challengeXDR, NETWORK_PASSPHRASE);
+  if (transactionSigned.source !== signingKey) {
+    throw new Error(`Invalid source account: ${transactionSigned.source}`);
```

Comment:
We are fetching the `signingKey` of the anchor from our own `.toml` url. I see no vector from which someone could send a malicious transaction for us to sign. 

This, in conjunction with the `0` sequence number check.